### ISSUE TITLE
Geolocation task not working when offline (no wifi/data)

### ISSUE THREAD

**Issue by DaviesHuang** [expo/expo#3565]
The startLocationUpdatesAsync is awesome! It works well on iOS, but in Android I have been experiencing no location updates when the phone is offline (no wifi or data, but GPS and location is still enabled).  

Would be greatful if you could let me know if this is an existing bug or expected but not documented behaviour.

### Environment

 Expo CLI 2.10.1 environment info:
    System:
      OS: Windows 8
    Binaries:
      npm: 6.3.0 - C:\Program Files\nodejs\npm.CMD

App target: Expo on Android

### Code
```
Expo.Location.startLocationUpdatesAsync(GEOFENCING_TASK_NAME, {
        accuracy: Expo.Location.Accuracy.BestForNavigation,
        timeInterval: 0, //milliseconds
        distanceInterval: 0, //meters
        showsBackgroundLocationIndicator: false
    });
```

```
TaskManager.defineTask(GEOFENCING_TASK_NAME, ({ data, error }) => {
        if (error) {
            console.warn(error.message);
            return;
        }
        if (data) {
            onReceiveLocationUpdate(data);
        }
    });
```

### Steps to Reproduce

(Write your steps here:)

1. Open the app with wifi on
2. Confirm that there is location update (logs in console)
3. Turn wifi off (Phone data was already off)
4. No more location updates being received

### Expected Behavior

(Write what you thought would happen.)
The app should continue receiving location update after turing wifi/data off.

### Actual Behavior

(Write what happened. Add screenshots!)
The app stops receiving location update after turing wifi/data off.

**Comment by superyarik** [comment]
also find this issue when trying to implement store locations to asyncstorage when offline, but there is no task callback firing then offline

**Comment by vadzzim** [comment]
The same story for me. But only want to add one detail. When you turn on wifi again. The task called multiple times. Something like... all tasks ware stored in a queue and paused while offline.

**Comment by ptbello** [comment]
I spent a few hours driving around to field test offline background location
I used a dummy app that triggers a local notification on every location update.
The notification body shows the recorded timestamp and location coords.

When online, evertything runs as expected.

When offline, the location is correctly recorded, but the task queue seems to be stuck, as in: notifications are not triggered.
When resuming an online connection, following a few minutes/kilometers of driving offline, all the notifications (dozens) are delivered at once.

Interestingly, the notifications are not delivered in chronological order: they arrive simultaneously, but are not displayed in the order dictated by the recorded timestamp.

It was possible to manually build a sample of chronologically ordered notifications (based on the recorded timestamp shown in the notification body), and trace the location coords on google maps.
Indeed, the recorded timestamp and locations accurately matched the route taken while offline.

To sum up: the problems seems to lie in the task queue, not in the location service.
When offline, the timestamp and location are recorded correctly but the associated task isn't triggered.
When back online, an accumulation of queued tasks is unlocked and floods the device.

This behaviour was tested with the dummy app installed both as an expo project (through the Expo app), and as a standalone built .apk. 
Also tested with device location service set both to "High accuracy" and to "Device only", with no significant difference.

**Comment by goskan93** [comment]
I have the same issue, I would be glad if it worked offline so much!

**Comment by wende60** [comment]
It seems that Location.getCurrentPositionAsync({}) also does not work if device is offline. However, adding Location.getCurrentPositionAsync({ enableHighAccuracy: true }) does the trick and works. Confusing...

**Comment by goskan93** [comment]
> It seems that Location.getCurrentPositionAsync({}) also does not work if device is offline. However, adding Location.getCurrentPositionAsync({ **enableHighAccuracy: true** }) does the trick and works. Confusing...

That did not help me...

**Comment by leoceolin** [comment]
> It seems that Location.getCurrentPositionAsync({}) also does not work if device is offline. However, adding Location.getCurrentPositionAsync({ **enableHighAccuracy: true** }) does the trick and works. Confusing...

doesn't work for me

**Comment by sharpcoding** [comment]
The following code (run in the scenario provided above 👆 i.e. no wifi/data):

```ts
import * as Location from "expo-location";
import { ToastAndroid } from "react-native";

//...

let { status } = await Location.requestForegroundPermissionsAsync();
if (status !== "granted") {
  return;
}
const providerStatus = await Location.getProviderStatusAsync();
const positionResult = await Location.getCurrentPositionAsync({
  accuracy: Location.Accuracy.Highest,
  mayShowUserSettingsDialog: true,
});
if (positionResult) {
  ToastAndroid.show(
    `Got accuracy: ${positionResult.coords.accuracy?.toFixed(2)} meters, gps: ${
      providerStatus.gpsAvailable
    }`,
    2000
  );
}
```
tested "in the field" (outside):
- gives accuracy of 1000m/2500m when run with newer Android Samsung device (Galaxy A51, Galaxy S10) 🙀
- gives accuracy of meters (4, 8, 12 etc) when run with older Android Samsung device (Galaxy S8)

The package versions:
- "expo-location": "^13.0.4"
- "react": "17.0.2"
- "react-native": "0.64.3"
- "expo": "43.0.3"
